### PR TITLE
[MI-3275]: Integrated config modal

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
Integrated config modal to edit and add config with configs.

### CODE REVIEW

File: webapp/src/containers/components/modals/deleteModal.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ function DeleteModal({visible, setVis, config, configIndex, onChange}: Props) {
                 </Modal.Header>
 
                 <Modal.Body>
-                    <p>{`Are you sure you would like to delete the configs for the team ${config[configIndex].TeamName}`}</p>
+                    <p>{`Are you sure you would like to delete the configs for ${config[configIndex].TeamName}`}</p>
```
Comment (avas27JTG) [inline webapp/src/containers/components/modals/deleteModal.tsx]:
```suggestion
                    <p>{`Are you sure you would like to delete the configs for ${config[configIndex].TeamName}`}?</p>
```

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 2]
```diff
@@ -85,6 +85,7 @@ const ExistingConfigTable = ({value, onChange}: Props) => {
                     {'Existing Configs'}
                 </div>
                 <div>
+                    {value.length > 0 &&
```
Comment (avas27JTG) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]:
```suggestion
                    {value.length &&
```
length can never be negative and `0` is already considered `false` by JS so, checking for just `value.length` means we are checking for `>= 1`

File: webapp/src/containers/components/modals/configModal.tsx [hunk 3]
```diff
@@ -9,24 +9,62 @@ import {OverlayTrigger, Tooltip, ToggleButton} from 'react-bootstrap';
...
 import './styles.css';
 
-// eslint-disable-next-line import/no-unresolved
-import {Configs} from 'types/plugin/common';
+import {Configs, Actions} from 'types/plugin/common';
 
 type Props = {
     visible: boolean;
-    setVis: React.Dispatch<React.SetStateAction<boolean>>;
+    setVisible: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]: fix according to previous prs

File: webapp/src/containers/components/modals/configModal.tsx [hunk 4]
```diff
@@ -9,24 +9,62 @@ import {OverlayTrigger, Tooltip, ToggleButton} from 'react-bootstrap';
...
+    const newAction: Actions[] = [
+    ];
+    const newConfig: Configs = {
+        teamName: '',
+        delayInSeconds: '',
+        message: [''],
+        includeGuests: '',
+        attachmentMessage: [''],
+        actions: newAction,
+    };
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]:
```suggestion
    };
    
```

File: webapp/src/containers/components/modals/configModal.tsx [hunk 5]
```diff
@@ -42,15 +80,51 @@ function ConfigModal(props: Props) {
...
+            setActionSuccessfullMessage(action.actionSuccessfullMessage);
+            setActionName(action.actionName);
+        } else {
+            const action = actionElement;
+            setActionTypesValue(action.actionType);
+            setActionDisplayName(action.actionDisplayName);
+            setChannelsAddedTo(action.channelsAddedTo);
+            setActionSuccessfullMessage(action.actionSuccessfullMessage);
+            setActionName(action.actionName);
+        }
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]: avoid using if and else both. Use if and return from it

File: webapp/src/containers/components/modals/configModal.tsx [hunk 6]
```diff
@@ -61,17 +135,113 @@ function ConfigModal(props: Props) {
...
+    const structureNewActions = () => {
+        actionElement.actionDisplayName = actionDisplayName;
+        actionElement.actionName = actionName;
+        actionElement.actionSuccessfullMessage = actionSuccessfullMessage;
+        actionElement.actionType = actionTypesValue;
+        actionElement.channelsAddedTo = actionChannelsAddedTo;
+        const l = existingConfig.actions?.push(actionElement);
+    };
+    const structureActions = () => {
+        const actions = existingConfig?.actions;
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]: use this directly

File: webapp/src/containers/components/modals/configModal.tsx [hunk 7]
```diff
@@ -132,245 +307,123 @@ function ConfigModal(props: Props) {
...
-                                                        />
-                                                    </svg>
-                                                </Button>
-                                            </OverlayTrigger>
-                                        </ButtonGroup>
-                                    </td>
-                                </tr>
-                            </tbody>
-                        </Table>
+                        {existingConfig?.actions && actionLength > 0 ? (
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/components/modals/configModal.tsx]:
```suggestion
                        {existingConfig?.actions && actionLength ? (
```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: webapp/src/containers/components/modals/configModal.tsx (hunk 3 above)
Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/configModal.tsx]: fixed in the following PRs

Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/modals/configModal.tsx]: #14

File: webapp/src/containers/components/tables/existingConfigTable.tsx [hunk 8]
```diff
@@ -42,162 +42,154 @@ const ExistingConfigTable = ({value, onChange}: Props) => {
...
+                                                            <Button
+                                                                onClick={() => handleView(i)}
+                                                            >
+                                                                <svg
+                                                                    xmlns='http://www.w3.org/2000/svg'
+                                                                    width='20'
+                                                                    height='20'
+                                                                    viewBox='0 0 24 24'
+                                                                    fill='none'
+                                                                    stroke='#333'
```
Comment (avas27JTG) [inline webapp/src/containers/components/tables/existingConfigTable.tsx]:
```suggestion
                                                                    stroke='#333333'
```
use hex

Comment (Kshitij-Katiyar) [reply webapp/src/containers/components/tables/existingConfigTable.tsx:107]: what is the difference between #000 & #000000 ?
Raghav suggests using #000

**Review by raghavaggarwal2308 — APPROVED** [review]

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**